### ISSUE TITLE
If an error is raised when consuming PowerTrack stream, there's no way to communicate it to the caller

### ISSUE THREAD

**Issue by j-bennet** [abh1nav/gnippy#12]
@abh1nav

I encountered a rare "Connection reset by peer" when consuming PowerTrack stream:

https://github.com/abh1nav/gnippy/blob/master/gnippy/powertrackclient.py#L75

The exception was thrown in `iter_lines`; unfortunately, I don't have the traceback available.

Because this error is happening within a worker thread, and there's no way to catch that, the calling thread also dies. I need a way to capture the error event and restart the connection to PowerTrack.

Suggestion:
- add an `exception_callback` parameter when creating `PowerTrackClient`
- all exceptions would be caught in worker thread's `run`
- if we have an `exception_callback` provided, that would be called, letting the caller know that a reconnect may be needed
- if no `exception_callback` is provided, re-raise.

If that sounds ok, I'd implement that.

Thoughts?

**Comment by joshfinnie** [comment]
@j-bennet 

I just received a similar error as you above (I think). Here is the traceback if that helps:

```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/site-packages/gnippy/powertrackclient.py", line 75, in run
    for line in r.iter_lines():
  File "/usr/local/lib/python2.7/site-packages/requests/models.py", line 699, in iter_lines
    for chunk in self.iter_content(chunk_size=chunk_size, decode_unicode=decode_unicode):
  File "/usr/local/lib/python2.7/site-packages/requests/models.py", line 657, in generate
    for chunk in self.raw.stream(chunk_size, decode_content=True):
  File "/usr/local/lib/python2.7/site-packages/requests/packages/urllib3/response.py", line 322, in stream
    for line in self.read_chunked(amt, decode_content=decode_content):
  File "/usr/local/lib/python2.7/site-packages/requests/packages/urllib3/response.py", line 467, in read_chunked
    self._update_chunk_length()
  File "/usr/local/lib/python2.7/site-packages/requests/packages/urllib3/response.py", line 414, in _update_chunk_length
    line = self._fp.fp.readline()
  File "/usr/local/lib/python2.7/socket.py", line 451, in readline
    data = self._sock.recv(self._rbufsize)
  File "/usr/local/lib/python2.7/ssl.py", line 734, in recv
    return self.read(buflen)
  File "/usr/local/lib/python2.7/ssl.py", line 621, in read
    v = self._sslobj.read(len or 1024)
error: [Errno 104] Connection reset by peer
```

**Comment by j-bennet** [comment]
@joshfinnie Exactly the one, thanks!

I'll submit a PR with the `exception_callback` I was talking about.

**Comment by abh1nav** [comment]
@j-bennet @joshfinnie Thanks for the fix :+1:  - it's available in v0.4.7 on PyPI.
